src/cargo/util/context/mod.rs

```diff
@@ -2030,18 +2030,22 @@ impl ConfigError {
         }
     }
 
+    fn is_missing_field(&self) -> bool {
+        self.error.to_string().contains("missing field")
+    }
```

Comment:
Could we have [serde tell us a field is missing](https://docs.rs/serde/latest/serde/de/trait.Error.html#method.missing_field) so we don't have to be doing string matching?

Reply:
Good idea. By Implementing the `missing_field` and return the `MissingField` error which we defined